File: filas/Fila.java
import java.util.ArrayList;
import java.util.List;

public class Fila {
    private List<Object> fila;

    public Fila() {
        this.fila = new ArrayList<>();
    }

    public boolean isEmpty() {
        return this.fila.isEmpty();
    }

    public void adiciona(Object item) {
        this.fila.add(item);
    }

    public Object remove() {
        if (!isEmpty()) {
            return this.fila.remove(0);
        } else {
            throw new IndexOutOfBoundsException("A fila está vazia.");
        }
    }

    public int tamanho() {
        return this.fila.size();
    }

    public Object primeiroElemento() {
        if (!isEmpty()) {
            return this.fila.get(0);
        } else {
            throw new IndexOutOfBoundsException("A fila está vazia.");
        }
    }

    public void clear() {
        this.fila.clear();
    }

    public boolean verificaPresenca(Object item) {
        return this.fila.contains(item);
    }

    public List<Object> copia() {
        return new ArrayList<>(this.fila);
    }
}

File: conteudos/filas/Fila.py
class Fila:
    
    def __init__(self) -> None:
        self.__fila = []
    
    def is_empty(self) -> bool:
        return len(self.__fila) == 0
    
    def adiciona(self, item) -> None:
        self.__fila.append(item)
        
    def remove(self) -> any:
        if not self.is_empty():
            return self.__fila.pop(0)
        else:
            raise IndexError("A fila está vazia.")        
                    
    def tamanho(self) -> int:
        return len(self.__fila)
    
    def primeiro_elemento(self) -> any:
        if not self.is_empty():
            return self.__fila[0]
        else:
            raise IndexError("A fila está vazia.")   
    
    def clear(self) -> None:
        self.__fila = []
        
    def verifica_presenca(self, item) -> bool:
        return item in self.__fila
    
    def copia(self) -> list:
        return self.__fila.copy()
    